src/cmdlib.sh

```diff
@@ -553,16 +553,20 @@ runcompose_tree() {
     else
         local tarball="${workdir}/tmp/repo/commit.tar"
         rm -f "${tarball}"
-        runvm_with_cache /usr/lib/coreos-assembler/compose.sh \
-            "${tarball}" "${composejson}" "$@"
-        if [ ! -f "${tarball}" ]; then
+        runvm_with_cache \
+            -chardev "file,id=tarout,path=${tarball}" \
+            -device "virtserialport,chardev=tarout,name=tarout" -- \
+            /usr/lib/coreos-assembler/compose.sh \
+                "/dev/virtio-ports/tarout" "$@"
+        if [ ! -s "${tarball}" ]; then
             return
         fi
-        local commit
-        commit=$(jq -r '.["ostree-commit"]' < "${composejson}")
         local import_repo="${workdir}/tmp/repo-import"
         rm -rf "${import_repo}" && mkdir "${import_repo}"
         tar -C "${import_repo}" -xf "${tarball}" && rm -f "${tarball}"
+        mv "${import_repo}/compose.json" "${composejson}"
+        local commit
+        commit=$(jq -r '.["ostree-commit"]' < "${composejson}")
```

Comment:
```suggestion
        local commit=$(jq -r '.["ostree-commit"]' < "${composejson}")
```

Reply:
wow, using `local` in front of a var assignment would do that?

should we drop a comment here for our future selves? This is the first thing I would do if I was modifying this code because look just a few lines above at `local import_repo=...`.

Reply:
Yeah, it's a weird bash gotcha. We do follow this pattern in many places in the codebase since we have ShellCheck in CI. I'm not sure it's worth commenting them all. I guess if you type it, a local `make shellcheck` will catch it (or if you don't run that, upstream CI will).